### ISSUE TITLE
Evaluate logical definitions using 'process_has_causal_agent ' 

### ISSUE THREAD

**Issue by pgaudet** [geneontology/go-ontology#20574]
Discussion of #17613 on the Ontology editors call highlighted that the relation 'process_has_causal_agent ' is probably incorrect. This relation was also questioned here: #20258.  

 - [ ]	intersection_of: process_has_causal_agent CHEBI:16480 ! nitric oxide
 - [ ]	intersection_of: process_has_causal_agent CHEBI:16670 ! peptide
 - [ ]	intersection_of: process_has_causal_agent CHEBI:16838 ! polyphosphate
 - [ ]	intersection_of: process_has_causal_agent CHEBI:17245 ! carbon monoxide
 - [ ]	intersection_of: process_has_causal_agent CHEBI:18059 ! lipid
 - [ ]	intersection_of: process_has_causal_agent CHEBI:33697 ! ribonucleic acid
 - [ ]	intersection_of: process_has_causal_agent CHEBI:48433 ! angiotensin
 - [ ]	intersection_of: process_has_causal_agent CHEBI:67194 ! cannabinoid
 - [ ]	intersection_of: process_has_causal_agent CL:0000084 ! T cell
 - [ ]	intersection_of: process_has_causal_agent CL:0000097 ! mast cell
 - [ ]	intersection_of: process_has_causal_agent CL:0000129 ! microglial cell
 - [ ]	intersection_of: process_has_causal_agent CL:0000233 ! platelet
 - [ ]	intersection_of: process_has_causal_agent CL:0000236 ! B cell
 - [ ]	intersection_of: process_has_causal_agent CL:0000451 ! dendritic cell
 - [ ]	intersection_of: process_has_causal_agent CL:0000542 ! lymphocyte
 - [ ]	intersection_of: process_has_causal_agent CL:0000545 ! T-helper 1 cell
 - [ ]	intersection_of: process_has_causal_agent CL:0000546 ! T-helper 2 cell
 - [ ]	intersection_of: process_has_causal_agent CL:0000623 ! natural killer cell
 - [ ]	intersection_of: process_has_causal_agent CL:0000624 ! CD4-positive, alpha-beta T cell
 - [ ]	intersection_of: process_has_causal_agent CL:0000738 ! leukocyte
 - [ ]	intersection_of: process_has_causal_agent CL:0000766 ! myeloid leukocyte
 - [ ]	intersection_of: process_has_causal_agent CL:0000767 ! basophil
 - [ ]	intersection_of: process_has_causal_agent CL:0000771 ! eosinophil
 - [ ]	intersection_of: process_has_causal_agent CL:0000775 ! neutrophil
 - [ ]	intersection_of: process_has_causal_agent CL:0000782 ! myeloid dendritic cell
 - [ ]	intersection_of: process_has_causal_agent CL:0000784 ! plasmacytoid dendritic cell
 - [ ]	intersection_of: process_has_causal_agent GO:0005768 ! endosome
 - [ ]	intersection_of: process_has_causal_agent GO:0005813 ! centrosome
 - [ ]	intersection_of: process_has_causal_agent GO:0005815 ! microtubule organizing center
 - [ ]	intersection_of: process_has_causal_agent GO:0005816 ! spindle pole body
 - [ ]	intersection_of: process_has_causal_agent GO:0005874 ! microtubule
 - [ ]	intersection_of: process_has_causal_agent GO:0005921 ! gap junction
 - [ ]	intersection_of: process_has_causal_agent GO:0009506 ! plasmodesma
 - [ ]	intersection_of: process_has_causal_agent GO:0022626 ! cytosolic ribosome
 - [ ]	intersection_of: process_has_causal_agent GO:0030139 ! endocytic vesicle
 - [ ]	intersection_of: process_has_causal_agent GO:0031931 ! TORC1 complex
 - [ ]	intersection_of: process_has_causal_agent GO:0031932 ! TORC2 complex
 - [ ]	intersection_of: process_has_causal_agent GO:0031982 ! vesicle
 - [ ]	intersection_of: process_has_causal_agent GO:0033281 ! TAT protein transport complex
 - [ ]	intersection_of: process_has_causal_agent GO:0038201 ! TOR complex
 - [ ]	intersection_of: process_has_causal_agent GO:0070062 ! extracellular exosome
 - [ ]	intersection_of: process_has_causal_agent GO:0098820 ! trans-synaptic protein complex
 - [ ]	intersection_of: process_has_causal_agent GO:0099073 ! mitochondrion-derived vesicle
 - [ ]	intersection_of: process_has_causal_agent GO:1903561 ! extracellular vesicle
 - [ ]	intersection_of: process_has_causal_agent PR:000000028 ! small GTPase
 - [ ]	intersection_of: process_has_causal_agent PR:000001245 ! rhodopsin
 - [ ]	intersection_of: process_has_causal_agent PR:000004716 ! brain-derived neurotrophic factor
 - [ ]	intersection_of: process_has_causal_agent PR:000011374 ! atrial natriuretic factor
 - [ ]	intersection_of: process_has_causal_agent PR:000012705 ! polymeric immunoglobulin receptor
 - [ ]	intersection_of: process_has_causal_agent SO:0000276 ! miRNA
 - [ ]	intersection_of: process_has_causal_agent SO:0000655 ! ncRNA

**Comment by ValWood** [comment]
Is this still valid?

**Comment by raymond91125** [comment]
From RO:
process has causal agent definition Inverse of 'causal agent in process'.
causal agent in process definition A relationship between a material entity and a process where the material entity has some causal role that influences the process.

#18197 globally replaced mediated_by with process_has_causal_agent.

**Comment by raymond91125** [comment]
**[Written by Claude]**

Two findings that bound what this ticket can deliver.

**1. The taxon-propagation goal is not reachable by changing the relation.** This ticket was opened the morning after RO#409 (`in taxon` chain for `process has causal agent`) was closed as "not general". But no process→continuant relation chains to `in taxon`: `occurs_in` (BFO:0000066) has chains only to/from `part of`, and `has input` only from `has part`. That matches what @balhoff reported in #17613 for all three relations there. RO's `in taxon` definition enumerates its only chain heads — *part of (reflexive), developmentally preceded by, derives from, secreted by, expressed* — all material-entity relations. So no relation swap will make `leukocyte mediated immunity` inherit Vertebrata. That needs a GO-local chain or direct TC assertion — the latter is what @raymond91125 proposed and @hattrill agreed to in #17613, and it never happened. Suggest we split it out and stop treating it as blocked here.

**2. `process has causal agent` is not a participation relation.** RO puts RO:0002608 under `causally related to`, not under `has participant`. So `X ≡ G and process_has_causal_agent some Y` does not assert that Y takes part in X — only that Y "has some causal role that influences" it. For `T cell mediated immunity` we mean the T cell *carries out* the process; the axiom doesn't say that, and as an equivalence it admits anything the T cell merely influences.

Context worth having: #18197 shows these axioms were never authored with RO:0002608 in mind. All 116 were a 2019 global find-and-replace of GOREL `mediated_by`, and "mediated by" was doing at least seven different jobs — effector cell, signal carrier, vehicle, machinery, scaffold, template, conduit.

So I'd restate the goal as: **triage the 104 terms by what the filler actually is, emit an "X-mediated process" pattern doc saying which relation each case wants, then migrate.** That gives the checklist an acceptance criterion it currently lacks. Triage table in the next comment.

**Comment by raymond91125** [comment]
**[Written by Claude]**

# Triage: 104 terms with a `process_has_causal_agent` logical definition

Grouped by **what the filler actually is**, since that — not the filler's ID space — is what determines the correct relation.

| # | Group | n | What the filler really is | Verdict | Priority |
|---|---|---|---|---|---|
| A | **Effector cell** | 36 | The cell type that *executes* the process | **Retype** — needs a participation relation | High |
| B | **Signal carrier / ligand** | 28 | The molecule that *carries* the signal | Keep pattern; fix fillers | Medium |
| C | **Vehicle / container** | 15 | The vesicle that *carries* the cargo | Keep — already endorsed | Low |
| D | **Machinery** | 10 | The complex that *performs* the process | **Keep** — genuinely agentive | None |
| E | **Scaffold / track** | 9 | The structure motors *run along* | **Fix** — axiom is false | High |
| D′ | **Receptor** | 2 | The receptor *transducing* the signal | Retype | Low |
| F | **Template** | 2 | The nucleic acid *copied from* | Retype | Low |
| G | **Conduit** | 2 | The junction transport *passes through* | Retype or demote | Low |

### A — Effector cell (36) · retype · high priority

All CL fillers: T cell (6), NK cell (4), mast cell (3), platelet (3), B cell (3), basophil (3), leukocyte (2), neutrophil (2), plus microglial cell, dendritic cell, myeloid DC, pDC, Th1, Th2, CD4+ αβ T cell, lymphocyte, myeloid leukocyte, eosinophil (1 each). Three sub-patterns: `X mediated immunity`/`cytotoxicity`, `X cytokine production`, `serotonin|histamine|protein secretion by X`.

This is the group #17613 was actually about, it is the largest, and it is where the causal-vs-participation gap bites hardest. `T cell mediated immunity ≡ immune system process and (causal agent some T cell)` is satisfied by any immune process a T cell influences. Candidate replacements present in the RO import: `has participant` (RO:0000057), `enabled by` (RO:0002333), `has primary input` (RO:0004009) — none obviously right, which is the substance of the design-pattern decision. Note `has active participant` is *not* in the import.

### B — Signal carrier / ligand (28) · keep pattern, fix fillers · medium

Nitric oxide (6), peptide zwitterion (5), trans-synaptic protein complex (4), lipid (3), cannabinoid (3), carbon monoxide (2), BDNF (2), plus polyphosphate, angiotensin, ANF. Almost entirely the SynGO trans-synaptic family.

Semantically the closest fit in the whole set, and the LDs do real classification work. Unsound as equivalences for the "influences" reason, but no competing class currently exploits it. Four concrete defects independent of that:

- **`PR:000004716` is now labelled *neurotrophic factor BDNF precursor form*** — GO:0099191 and GO:0099183 are defined by the precursor, not mature BDNF. Substantive error.
- **Neuropeptide terms use `CHEBI:60466 peptide zwitterion`** — any peptide, five terms.
- **Endocannabinoid terms use `CHEBI:67194 cannabinoid`** — includes phytocannabinoids, three terms.
- **`GO:0099543 trans-synaptic signaling by soluble gas` has no LD at all** while every child has one; NO and CO are hung under it by hand.

Nitric oxide is the best-behaved row in the entire 104 — exact filler, label/def/LD aligned. Two of its six are SynGO transitional terms already marked for obsoletion.

### C — Vehicle / container (15) · keep · low

Vesicle (10), extracellular exosome (2), exocytic vesicle, extracellular vesicle, endosome. The exocytosis axiom was added deliberately in #22569 after discussion with @balhoff and @cmungall, and in #32166 the causal-agent differentia on `vesicle-mediated intercellular transport` was the one axiom in that family that *didn't* misbehave. Leave alone.

### D — Machinery (10) · keep · no action

TORC1/TORC2/TOR signaling, `protein transport by the Tat complex`, `cytoplasmic translation` (cytosolic ribosome), `small GTPase-mediated signal transduction`, and the four microtubule-nucleation terms (centrosome, MTOC ×2, spindle pole body). These are true causal agents — the ribosome really does perform translation, the MTOC really does nucleate. If the relation survives anywhere, it survives here.

### E — Scaffold / track (9) · fix · high priority

Microtubule (8): `cilium movement`, `microtubule-based process`, `microtubule-based transport`, `microtubule-based protein transport`, `chromosome movement towards spindle pole`, `establishment of mitochondrion localization, microtubule-mediated`, and two MT-polymerization transport terms. Plus actin filament (1): `actin filament-based process`.

#20258's case, and the only group where the axiom is plainly *false* rather than merely too weak. `GO:0003341 cilium movement` carries the double error @krchristie identified — MT as causal agent, plus `results_in_movement_of cilium` on a term named cilium movement. Caveat: `GO:0007017` and `GO:0030029` are cytoskeleton branch roots, so this is the highest-blast-radius group; reason before and after.

### D′, F, G — small tail (6) · low

- **Receptor (2):** `G protein-coupled opsin signaling pathway` (rhodopsin), `immunoglobulin transcytosis … mediated by polymeric immunoglobulin receptor`.
- **Template (2):** `DNA-templated transcription` (DNA), `RNA-templated transcription` (RNA) — the nucleic acid is copied from, not an agent.
- **Conduit (2):** `gap junction-mediated intercellular transport`, `plasmodesmata-mediated intercellular transport` — channels, though #32166 notes both are well-behaved in practice.

---

**Suggested order of work:** E first (smallest defensible fix, clearest error, but reason carefully); then A (largest, and the one that needs the pattern decision made); B's four filler defects can go in independently of any relation decision, since they are wrong under any reading. C and D need nothing.

**Comment by raymond91125** [comment]
**[Written by Claude]**

@pgaudet @balhoff — a decision is needed to unblock #32553, and I think it is narrower and lower-stakes than the triage above implied. Restating it in the terms that actually matter.

## The question is not "is `occurs_in` true"

Both candidate relations state something biologically true of `T cell cytokine production`. The process is located in the T cell, and the T cell carries it out. Biology does not choose between them. The question is which claim GO wants the axiom to make: **location, or agency.**

That matters because I initially framed this group as needing a participation relation, and `occurs_in` (BFO:0000066) is not one. On the merits that framing is right — "T cell cytokine production" names something the T cell *does*, and `occurs_in` says only where it happens. `enabled by` would be the more faithful claim.

## But GO has already answered this, 129 times

The `genus + occurs_in some <cell type>` shape is not a new proposal. It is what GO does, including inside the very families at issue:

| term | differentia |
|---|---|
| `GO:0010934` macrophage cytokine production | `occurs_in CL:0000235` |
| `GO:0032762` mast cell cytokine production | `occurs_in CL:0000097` |
| `GO:0002561` basophil degranulation | `occurs_in CL:0000767` |
| `GO:0002576` platelet degranulation | `occurs_in CL:0000233` |
| `GO:0002514` B cell tolerance induction | `occurs_in CL:0000236` |
| `GO:0002517` T cell tolerance induction | `occurs_in CL:0000084` |
| `GO:0002519` natural killer cell tolerance induction | `occurs_in CL:0000623` |
| `GO:0010931` macrophage tolerance induction | `occurs_in CL:0000235` |

Two consequences worth being explicit about.

**The `X cytokine production` family is split today.** Macrophage and mast cell use `occurs_in`; B cell, T cell, NK cell, dendritic cell and the Th1/Th2 terms use `process_has_causal_agent`. Same biology, same shape, two different relations, purely by authorship history.

**The degranulation pair disposes of the one real objection.** The obvious complaint about `occurs_in` on secretion terms is that secretion moves material out of the cell, so the process is not wholly located within it. `GO:0002561` and `GO:0002576` are the `part_of` parents of six of the seven secretion terms in #32553, and both are already defined as `regulated exocytosis and occurs_in some <cell>`. GO made that call already; parent and child simply disagree with each other right now.

## The cost asymmetry

If the answer is "location, consistent with existing practice", #32553 merges and the split families become uniform.

If the answer is "agency, this shape should use a participation relation", then the work is not retyping the 16 terms in #32553. It is revisiting all **145** `occurs_in` + cell-type logical definitions, since they all make the same kind of claim. That is a legitimate direction, but it is a much larger project than this ticket, and it needs `has active participant` or `enabled by` brought into the import first — neither is currently in `go-edit.obo`.

So the practical question is:

> Should `X cytokine production` and `X secretion by X` assert **location** (`occurs_in`), consistent with the 129 existing axioms of that shape — or **agency**, which would mean revisiting those 129 as well?

One line either way unblocks #32553.

## What #32553 actually does

16 terms, one line each, `process_has_causal_agent` to `occurs_in`: 9 `X cytokine production` and 7 `X secretion by X`. Definitions, genus and labels untouched.

Verified with an ELK reason over the full import closure before and after, comparing every subclass link with a GO subject: 76,680 links both sides, none gained, none lost. `make travis_test` passes. Two `is_a` links are asserted by hand because their parents still use the old relation; both are marked in the file with a `term_tracker_item` back to this ticket and should be dropped when the rest of the group is settled.

Deliberately excluded, since `occurs_in` would be false for them: `X mediated immunity` (12 grouping terms broader than one cell) and `X mediated cytotoxicity` (5, which span killer and target cell). The 3 `X tolerance induction` terms are held back only because they extend the same undecided question — GO already models that family with `occurs_in`, so they are the obvious follow-up.

The other 68 `process_has_causal_agent` logical definitions (non-CL fillers) are untouched and unaffected by this decision.

**Comment by raymond91125** [comment]
**[Written by Claude]**

Status after group A, and a correction to the plan in the triage above.

## Group A - done (36 axioms)

- **19 retyped to `occurs_in`** - cytokine production (9), secretion (7) in #32553; tolerance induction (3) in #32554. `occurs_in` with a CL filler was already GO practice in these families: `GO:0010934 macrophage cytokine production`, `GO:0032762 mast cell cytokine production`, `GO:0002561 basophil degranulation`, `GO:0002576 platelet degranulation` and the four `X tolerance induction` terms all used it before this work.
- **17 demoted to `relationship:`** - `X mediated immunity` (12) and `X mediated cytotoxicity` (5) in #32554. No relation is correct for these as an *equivalence*: the immunity terms are grouping classes broader than one cell, and cytotoxicity spans killer and target.

Both PRs measured with a full-closure ELK reason (76,680 `rdfs:subClassOf` pairs with a GO subject): zero lost, zero gained. `GO:0002443` and `GO:0002444` now carry comments recording that the absence of a logical definition is deliberate.

## The relation itself is not the defect

Worth recording, because it changes what this ticket should aim at. **`RO:0002218 'has active participant'` is obsolete, and RO redirects it to `RO:0002500 'causal agent in process'`** - whose inverse is `RO:0002608`, the relation in question:

```
RO:0002218  obsolete has active participant
            IAO:0100001 (replaced by) -> RO:0002500 causal agent in process
RO:0002500  owl:inverseOf -> RO:0002608 process has causal agent
```

So RO deliberately retired the participation-flavoured agency relation and moved those semantics into the causal branch. `process_has_causal_agent` is RO's current sanctioned way to say a material entity is the agent of a process, and swapping it for `has_participant` ("somehow involved") would be a downgrade - in `T cell mediated cytotoxicity` the target cell participates as much as the killer.

The defect in group A was the use of these axioms as **equivalences**, not the relation. That is what both PRs fixed.

The one criticism that survives: `RO:0002500` is defined as "some causal role that influences the process", which an inhibitor also satisfies. That is an RO issue with no better-defined alternative in the current release.

## Group B (28) - demotion is actively harmful; keep the axioms

The triage above said "keep pattern; fix fillers". I tested that rather than assuming, by demoting all 28 and reasoning. Result: **34 links lost and 52 gained, and the gains are false.**

The cause is that the causal-agent differentia is the only thing distinguishing siblings in the SynGO lattice. Every term shares the same genus and the same structural differentiae - direction (`has_start_location`/`has_end_location`) and modulation (`regulates`) - and differs only in the ligand. Remove it and four terms become character-for-character identical:

```
GO:0099183  by BDNF, modulating            |
GO:0099550  (generic) modulating           |  all become
GO:0099552  by lipid, modulating           |  trans-synaptic signaling
GO:0099553  by endocannabinoid, modulating |  and regulates some chemical synaptic transmission
```

The reasoner then makes them mutually equivalent, which is where nonsense like `trans-synaptic signaling by BDNF` ⊑ `trans-synaptic signaling by soluble gas` comes from. That is worse than the over-inclusion the ticket set out to fix.

There is also nowhere to migrate to. `process_has_causal_agent` is the only relation GO uses for a chemical differentia in any signaling term (all 24). `has_primary_input` is wrong by definition - RO:0004009 requires the process to "modify, consume, or transform" the chemical, and a signal molecule is none of those.

**Verdict: group B keeps the relation.** Its real defects are the fillers:

- **Fixed:** `CHEBI:67194 cannabinoid` -> `CHEBI:67197 endocannabinoid` on 3 terms (#32560). All three definitions already said "endocannabinoid ligand"; `cannabinoid` also covers phytocannabinoids.
- **Blocked - no ChEBI target:** the 5 neuropeptide terms use `CHEBI:60466 peptide zwitterion` (any peptide). **ChEBI has no class labelled "neuropeptide"**, only receptor agonists and antagonists. Needs a ChEBI request.
- **Blocked - ChEBI classification gap:** `GO:0099543 trans-synaptic signaling by soluble gas` has no logical definition. The obvious differentia, `CHEBI:138675 gas molecular entity`, would not work: **nitric oxide is not classified under it** (ChEBI places `CHEBI:16480` under nitrogen oxide / inorganic radical / reactive nitrogen species) while `CHEBI:17245 carbon monoxide` is a direct child. An axiom there would silently miss the NO children.
- **Blocked - needs a curator decision:** `PR:000004716` is now labelled *"neurotrophic factor BDNF precursor form"*, so `GO:0099183` and `GO:0099191` are defined by the precursor. I could not resolve the mature-form PR id (absent from `imports/pr_import.owl`, PR is not served by OLS4). Beyond the lookup there is a scoping question: proBDNF and mature BDNF both signal trans-synaptically, via p75NTR and TrkB, with opposing effects on plasticity - so "trans-synaptic signaling by BDNF" may mean the mature form or both.

### One candidate worth a future RO ticket

`RO:0002019 'has ligand'` is the only relation found that would be a genuine improvement for group B - *"a chemical entity ... that carries information between cells or compartments"*, exactly these terms' semantics, and far narrower than "has some causal role that influences".

It is not usable as it stands. Its definition says it holds "between a receptor and a chemical entity" (material -> material), but its only formal axiom is `subPropertyOf RO:0002233 has input`, whose domain is `BFO:0000015 process`. Definition and axiomatization disagree. `has input` also entails that "the state of c is modified during p", which is false for a ligand that binds and diffuses away. It is already seeded in `imports/ro_terms.txt` but has no typedef in `go-edit.obo`.

Not proposing to act on it here - flagging it as the one principled replacement available if RO clarifies the domain.

## Groups C and D (25) - no action, as triaged

Vesicle/vehicle (15) and machinery (10). The exocytosis axiom was added deliberately in #22569; the ribosome really does perform translation and the MTOC really does nucleate.

Separately, `GO:0098876` carried the same differentia as both an `intersection_of` and a `relationship:` - one redundant line, removed in #32559.

## Next: group E (9)

The remaining case where the axiom is plainly **false** rather than merely weak - microtubule (8) and actin filament (1) as "causal agent" of transport they are the track for. `results_in_transport_along` already exists as a typedef, so this needs no new relation and no upstream change.

## Revised scope for this ticket

This ticket cannot be closed by eliminating `process_has_causal_agent`. On the evidence it should close as: group A resolved, group E fixed, group B kept with filler corrections, groups C and D untouched, with the ChEBI/PR/RO items raised as their own tickets.

**Comment by raymond91125** [comment]
**[Written by Claude]**

So #20574 can't close as "eliminate the relation". It should close as: group A resolved, group E cleaned, group B and C/D deliberately kept, with the ChEBI/PR/RO items and the duplicate pairs spun out as their own tickets. That conclusion is recorded on the issue, but it reverses the ticket's framing, so it's worth @pgaudet or @balhoff confirming rather than me closing it on my own analysis.
